jobbergate-api/jobbergate_api/apps/job_submissions/schemas.py

```diff
@@ -60,9 +60,218 @@
         description="The report message received from cluster-agent when a job submission is rejected",
         example="Unrecognized SBATCH arguments",
     ),
+    execution_parameters=MetaField(
+        description="The parameters to be passed to the job submission",
+        example={"name": "job-submission-name", "comment": "I am a comment"},
+    ),
 )
 
 
+class JobProperties(BaseModel, extra=Extra.forbid):
+    """
+    Specialized model for job properties.
+
+    See more details at: https://slurm.schedmd.com/rest_api.html#v0.0.36_job_properties
+    """
+
+    account: Optional[str] = Field(
+        description="Charge resources used by this job to specified account.",
+    )
+    account_gather_freqency: Optional[str] = Field(
+        description="Define the job accounting and profiling sampling intervals.",
+    )
+    argv: Optional[List[str]] = Field(description="Arguments to the script.")
+    array: Optional[str] = Field(
+        description=(
+            "Submit a job array, multiple jobs to be executed with identical parameters. "
+            "The indexes specification identifies what array index values should be used."
+        )
+    )
+    batch_features: Optional[str] = Field(description="features required for batch script's node")
+    begin_time: Optional[str] = Field(
+        description=(
+            "Submit the batch script to the Slurm controller immediately, like normal, "
+            "but tell the controller to defer the allocation of the job until the specified time."
+        )
+    )
+    burst_buffer: Optional[str] = Field(description="Burst buffer specification.")
+    cluster_constraints: Optional[str] = Field(
+        description=(
+            "Specifies features that a federated cluster must have to have a sibling job submitted to it."
+        )
+    )
+    comment: Optional[str] = Field(description="An arbitrary comment.")
+    constraints: Optional[str] = Field(description="node features required by job.")
+    core_specification: Optional[int] = Field(
+        description=(
+            "Count of specialized threads per node reserved by the job for system "
+            "operations and not used by the application."
+        )
+    )
+    cores_per_socket: Optional[int] = Field(
+        description=(
+            "Restrict node selection to nodes with at least the specified number of cores per socket."
+        )
+    )
+    cpu_binding: Optional[str] = Field(description="Cpu binding")
+    cpu_binding_hint: Optional[str] = Field(description="Cpu binding hint")
+    cpu_frequency: Optional[str] = Field(
+        description=(
+            "Request that job steps initiated by srun commands inside this sbatch "
+            "script be run at some requested frequency if possible, on the CPUs "
+            "selected for the step on the compute node(s)."
+        ),
+    )
+    cpus_per_gpu: Optional[str] = Field(description="Number of CPUs requested per allocated GPU.")
```

Comment:
you sure this is a string?

Reply:
According to slurm's documentation, it is:
<img width="244" alt="image" src="https://user-images.githubusercontent.com/37457501/204589160-3a31c21b-deef-41e2-8010-f94c924df93d.png">

It is also supported by the slurm API's JSON schema:

```
"cpus_per_gpu": {
    "type": "string",
    "description": "Number of CPUs requested per allocated GPU."
},
```

Thanks for pointing it out, I agree `int` would make more sense, I'll keep an eye on this when working on the next steps of this task. This was the first sub-task, and I need to work on a couple more to be able to perform integration tests.